docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,169 @@
+:EP: 38
+:Title: Action Pacing
+:Authors:
+    - David Ramirez <davramir@fiu.edu>;
+:Issued Date: 2024-05-16
+:Status: Pending
+:Type: Standards Track
+
+*********************
+EP038 - Action Pacing
+*********************
+
+########
+Abstract
+########
+
+This document is intended on a guide for both NApp Developers
+and Network Operators for how to implement and use Pacing within Kytos.
+Pacing is intended as a mechanism to control
+how often certain actions will be executed.
+This differs from rate limiting, as the action is not cancelled,
+and instead the caller is temporarily paused until the action
+is allowed to be executed again.
+For an action to be paced, it needs to be instrumented
+by developers to support pacing.
+
+##########
+Motivation
+##########
+
+The purpose of adding in pacing to Kytos is
+help maintain system stability by preventing
+excessive throughput.
+It could occur that sending too many flow mods
+or too many DB requests may result in
+connection failures, and therefore system
+instability.
+To prevent such scenarios,
+pacing of actions which are sensitive to high
+throuhput should be implemented.
+
+#############
+Specification
+#############
+
+This section describes how to set the pace of actions,
+intended for network operators,
+and how to add pacing to an action,
+intended for developers.
+
+Setting A Pace
+==============
+
+For network operators,
+setting the pace of an action can be done through
+a NApp's ``action_paces`` setting.
+The ``action_paces`` setting is a dictionary,
+with keys representing the action to be paced,
+and values being the rate to pace them to.
+The pacing value is a string formatted as
+``{max_concurrent}/{time_period}{time_unit}``,
+where ``max_concurrent`` is the max amount of times
+the action is allowed to be executed
+within the ``time_period`` using the given ``time_unit``.
+Here's an example:
+
+.. code-block:: python
+
+    action_paces = {
+        "send_flow_mod": "50/2seconds",
+    }
+
+How the ``time_period`` is interpreted is controlled by the pacing strategy.
+Pacing strategy can be set using the ``pacing_strategy`` option in ``kytos.conf``.
+Available strategies include:
+
+ - `fixed_window` - Actions are paced in a fixed window,
+   where actions are paced in fixed blocks of time
+   that reset after the amount of time in the refresh period has passed.
+ - `elastic_window` - Actions are paced in an elastic window,
+   which acts like a fixed window, except if the pace is exceeded
+   then the window refresh period is doubled.
+ - `moving_window` - Actions are paced in a moving window,
+   where actions are paced in a dynamic block of time
+   that trails behind the current time.
+
+
+
+Pacing an Action
+================
+
+In order to add pacing functionality to an action,
```

Comment:
It's implicitly clear that this is a producer scoped pacer. If you could briefly explain about it a bit and where to usually place it, that'd be great for developers to understand even more what it does in this current iteration and what not to do. If we can have a paragraph stating that it's a pacer for producer and it should pace as close to the source that needs to be paced that's already helpful. 

I'm thinking about that case where we have 3 producers, `mef_eline`, `telemetry_int`, and `of_multi_table` and how each of them will not over a specific rate when publishing the events to `flow_manager` (which also might have another pacer to pace by dpid as you exemplified below).

Reply:
Added in some tips for where to implement, depending on the developer's intentions.